core/core-impl/src/main/java/ai/timefold/solver/core/impl/heuristic/selector/move/composite/UnionMoveSelectorFactory.java

```diff
@@ -22,8 +23,24 @@ public UnionMoveSelectorFactory(UnionMoveSelectorConfig moveSelectorConfig) {
     @Override
     protected MoveSelector<Solution_> buildBaseMoveSelector(HeuristicConfigPolicy<Solution_> configPolicy,
             SelectionCacheType minimumCacheType, boolean randomSelection) {
-        List<MoveSelector<Solution_>> moveSelectorList = buildInnerMoveSelectors(config.getMoveSelectorList(),
-                configPolicy, minimumCacheType, randomSelection);
+        List<MoveSelectorConfig> moveSelectorConfigList = new LinkedList<>(config.getMoveSelectorList());
+        if (configPolicy.getNearbyDistanceMeterClass() != null) {
+            for (MoveSelectorConfig selectorConfig : config.getMoveSelectorList().stream()
+                    .filter(MoveSelectorConfig::acceptNearbySelectionAutoConfiguration).toList()) {
+                if (selectorConfig.hasNearbySelectionConfig()) {
+                    throw new IllegalArgumentException(
```

Comment:
Maybe explain _why_ in a comment?

The problem with cartesian is that if there are two selectors, it will have A*B moves. So if there are four selectors (2 non-nearby + 2 nearby), it will be A*B*C*D moves.

Whereas union will always be A+B+C+D.